### PR TITLE
Preparing for productization

### PR DESCRIPTION
**Pull request by shiftylogic**
- new trimmed solution file.
- Removed System.Slices.
- Created a new System.Buffers.Primitives assembly.
- Moved memory related structs (Memory<T>, OwnedMemory, etc) into System.Buffers.Primitives.
- Moved span extensions that are being used to System.Buffers.Primitives.
- Moved anything else in System.Slices into System.Buffers.Experimental for now.

@KrzysztofCwalina @ahsonkhan

### CODE REVIEW

**Comment by ahsonkhan** [comment]
> new trimmed solution file.

It is hard to discern the diff between this new sln file and the old one. Can you rename it in a separate commit or some other workaround so we can see the diff first?

**Comment by shiftylogic** [comment]
This is a "new" file. The old one is still there. This just makes it so you don't have to load and build everything whenever you want to work on the production stuff.

File: scripts/build.ps1 [hunk 1]
```diff
@@ -22,7 +22,7 @@ $dotnetExePath="$PSScriptRoot\..\dotnet\dotnet.exe"
 
 if ($Restore -eq "true") {
     Write-Host "Restoring all packages"
-    Invoke-Expression "$dotnetExePath restore"
+    Invoke-Expression "$dotnetExePath restore corefxlab.sln"
```
Comment (ahsonkhan) [inline scripts/build.ps1]: Is this necessary now because we have two sln files?

Comment (shiftylogic) [inline scripts/build.ps1]: yes

**Comment by ahsonkhan** [comment]
You are missing a change in the PerfHarness.cs
https://github.com/dotnet/corefxlab/blob/c1a0338149df653b50948a7ef0c1ff32f150e829/scripts/PerfHarness/PerfHarness.cs#L52

Please run the benchmarks as well as part of the testing for this change.

**Comment by ahsonkhan** [comment]
LGTM

**Comment by KrzysztofCwalina** [comment]
Assuming there are no changes to the large green files that seem to be just moved, it looks good.

**Comment by ahsonkhan** [comment]
Part of https://github.com/dotnet/corefxlab/issues/1314

**Pull request merged**